src/LogicTasks/Syntax/SubTreeSet.hs

```diff
@@ -124,26 +139,30 @@ partialGrade' SubTreeInst{..} fs
 
 
 completeGrade
-  :: (OutputCapable m, MonadIO m)
+  :: (OutputCapable m, MonadIO m, Alternative m)
   => FilePath
   -> SubTreeInst
   -> Delayed [FormulaAnswer]
-  -> LangM m
+  -> Rated m
 completeGrade path inst = completeGrade' path inst `withDelayedSucceeding` parser
 
 completeGrade'
-  :: (OutputCapable m, MonadIO m)
+  :: (OutputCapable m, MonadIO m, Alternative m)
   => FilePath
   -> SubTreeInst
   -> [FormulaAnswer]
-  -> LangM m
-completeGrade' path SubTreeInst{..} sol = refuseIfWrong $ do
-  unless partOfSolution $ do
-    instruct $ do
-        english "Your solution is incorrect."
-        german "Ihre Lösung ist falsch."
-
-  when showSolution $ indent $ do
+  -> Rated m
+completeGrade' path SubTreeInst{..} sol = reRefuse
+  (extendedMultipleChoice
+    (MinimumThreshold (1 % minInputTrees))
```

Comment:
This might as well be `0` now, right?

Reply:
Yes, we could change that. But wouldn't that also change how the submission is shown on the task statistics page in Autotool? With `MinimumThreshold 0` every submission would be marked as correct (green). On the other side, keeping it like this would mark it as semantically wrong (red). 

If this is the case, I would personally prefer the current implementation.

Reply:
... considering a submission with 0 Points.

Reply:
Or I'm confused again in a different way. :smile:

The point (for Autotool color display) is whether or not `refuse` gets called?

Reply:
> If the threshold is 0.5 and a student would get 0.3 points, what happens? Do the actually get 0 points? 

The result of `completeGrade` would be `Nothing` here. This should lead to 0 points (my assumption). 
If the student got 0.7 points, it would be `Just (7 % 10)`.

My guess would be:
- submissions with `Nothing` are marked red, showing the reason of failure (Syntax or Semantics)
- submissions with `Just _` are marked green, showing the received amount of points

This would be the same behavior as with the usage of `reject` (using `reject` during grading leads to `Nothing`).